pkg/api/pod/util.go

```diff
@@ -1050,19 +1057,25 @@ func nodeTaintsPolicyInUse(podSpec *api.PodSpec) bool {
 
 // hostUsersInUse returns true if the pod spec has spec.hostUsers field set.
 func hostUsersInUse(podSpec *api.PodSpec) bool {
-	if podSpec != nil && podSpec.SecurityContext != nil && podSpec.SecurityContext.HostUsers != nil {
-		return true
-	}
-
-	return false
+	return podSpec != nil && podSpec.SecurityContext != nil && podSpec.SecurityContext.HostUsers != nil
 }
 
 func supplementalGroupsPolicyInUse(podSpec *api.PodSpec) bool {
-	if podSpec != nil && podSpec.SecurityContext != nil && podSpec.SecurityContext.SupplementalGroupsPolicy != nil {
-		return true
+	return podSpec != nil && podSpec.SecurityContext != nil && podSpec.SecurityContext.SupplementalGroupsPolicy != nil
+}
+
+func podObservedGenerationTrackigInUse(podStatus *api.PodStatus) bool {
+	if podStatus == nil {
+		return false
 	}
 
-	return false
+	for _, condition := range podStatus.Conditions {
+		if condition.ObservedGeneration != 0 {
+			return true
+		}
+	}
+
+	return podStatus.ObservedGeneration != 0
```

Comment:
nit: do this check first before iterating over all the conditions, just to check the most likely field that would be set before an entire list from random controllers that might be old and not using this field yet

```go
if podStatus.ObservedGeneration != 0 {
  return true
}
for _, condition := range podStatus.Conditions {
 ...
}
return false
```

Reply:
done